src/server/handshake.js

```diff
@@ -8,6 +8,12 @@ module.exports = function (client, server, { version }) {
     client.serverPort = packet.serverPort
     client.protocolVersion = packet.protocolVersion
 
+    if (packet.nextState === 1) {
+      client.state = states.STATUS
+    } else if (packet.nextState === 2) {
+      client.state = states.LOGIN
+    }
+
```

Comment:
You need to assign `client.version` before `client.state` since `version` is used to build the serializer

Reply:
Or reassign client.state to rebuild the serializer :thinking:

What do you think @rom1504 ?